WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs

```diff
@@ -0,0 +1,219 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reactive.Linq;
+using Avalonia.Controls;
+using Avalonia.Controls.Models.TreeDataGrid;
+using Avalonia.Controls.Templates;
+using DynamicData;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.TreeDataGrid;
+using WalletWasabi.Fluent.ViewModels.CoinControl.Core;
+using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Cells;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Headers;
+
+namespace WalletWasabi.Fluent.ViewModels.CoinControl;
+
+public partial class CoinSelectorViewModel : ViewModelBase, IDisposable
+{
+	[AutoNotify] private IReadOnlyCollection<SmartCoin> _selectedCoins = ImmutableList<SmartCoin>.Empty;
+
+	public CoinSelectorViewModel(WalletViewModelBase walletViewModel, IEnumerable<SmartCoin> initialCoinSelection)
+	{
+		var pockets = walletViewModel.Wallet.GetPockets();
+		var pocketItems = CreatePocketItems(pockets);
+
+		SyncSelectedItems(pocketItems, initialCoinSelection);
+		CollapseUnselectedPockets(pocketItems);
+
+		var pocketColumn = PocketColumn();
```

Comment:
Why we need local var pocketColumn ?

Reply:
Leftover from previous version where we sorted by that column 😅